core/storage/btree.rs

```diff
@@ -1423,25 +1479,26 @@ impl BTreeCursor {
             write_varint_to_vec(record_buf.len() as u64, cell_payload);
         }
 
-        let max_local = self.max_local(page_type.clone());
+        let payload_overflow_threshold_max = self.payload_overflow_threshold_max(page_type.clone());
         log::debug!(
-            "fill_cell_payload(record_size={}, max_local={})",
+            "fill_cell_payload(record_size={}, payload_overflow_threshold_max={})",
             record_buf.len(),
-            max_local
+            payload_overflow_threshold_max
         );
-        if record_buf.len() <= max_local {
+        if record_buf.len() <= payload_overflow_threshold_max {
             // enough allowed space to fit inside a btree page
             cell_payload.extend_from_slice(record_buf.as_slice());
             cell_payload.resize(cell_payload.len() + 4, 0);
             return;
         }
         log::debug!("fill_cell_payload(overflow)");
 
-        let min_local = self.min_local(page_type);
-        let mut space_left = min_local + (record_buf.len() - min_local) % (self.usable_space() - 4);
+        let payload_overflow_threshold_min = self.payload_overflow_threshold_min(page_type);
+        let mut space_left = payload_overflow_threshold_min
```

Comment:
I don't understand this calculation and it should be commented

Reply:
I don't remember right now. Might need to look at it again. There are things that are basically 1:1 copies of what sqlite does that's why names are like that

Reply:
I can try to find the places in sqlite code that these are direct c-to-rust translations of and attach them as comments

Reply:
also we can keep calling these `max_local` and `min_local` if it helps in referencing back to sqlite later. i just found those names confusing and tried to come up w something else, although it's difficult to translate these into concise variable names:

```
max_local = maximum payload size before the content must spill to overflow pages
min_local = minimum payload size when the content is ALLOWED to spill to overflow pages
```

Reply:
Added a reference to sqlite source in https://github.com/tursodatabase/limbo/pull/539/commits/c417fe788083f2dfa20970c9954facaabd9cf654